### DISCUSSION TITLE
Problem when using ASCII characters in R code

### DISCUSSION THREAD

**Discussion by maxl-amberg** [ThinkR-open/golem#659] [Q&A] [answered]
Hi,
I've got one question regaring one of my apps which runs perfectly fine when I use "run_dev.R" stored in the "dev" folder, but which always renders the following WARNING when I run devtools::check() in "03_deploy.R":
_Found the following file with non-ASCII characters:
    **app_ui.R**
    **mod_dataviz.R**
  Portable packages must use only ASCII characters in their R code,
  except perhaps in comments.
  Use \uxxxx escapes for other characters._

Since I would like to publish the app in German, my dataset contains variables that in turn contain _Umlaute_ such as Ä, Ö, Ü, ä, ö, and ü (which all show up in both app_ui.R and mod_dataviz.R). @ColinFay : Despite having specified UTF-8 in the global options, do I have to fix the warning concerning these non-ASCII characters by replacing Ä, Ö, Ü, ä, ö, and ü in whatever way by \u00c4, \u00d6, \u00dc, \u00e4, \u00f6, and \u00fc respectively?

Note that replacing "RO-1: Ohne \nRückerstattung" by "RO-1: Ohne \nR"\u00fcckerstattung" in **_mod_dataviz.R_** solves the first warning regarding **_mod_dataviz.R_** but renders the following additional warning:
_Found the following file with non-ASCII characters:
    **app_ui.R**
  Portable packages must use only ASCII characters in their R code,
  except perhaps in comments.
  Use \uxxxx escapes for other characters.
Warning in file 'R/mod_dataviz.R':
    unable to translate 'RO-1: Ohne R<U+00FC>ckerstattung' to native encoding_

Could replacing every Ä, Ö, Ü, ä, ö, and ü in **_app_ui.R_** and **_mod_dataviz.R_** work if instead done in the following fashion(?):
stringr::str_c("RO-1: Ohne \nR", "\u00fc", "ckerstattung") or
paste0("RO-1: Ohne \nR", "\u00fc", "ckerstattung")

Note that I actually do not want to sent the app/package to CRAN or to another package manager later on - I only want to deploy it e.g. via shinyapps.io or rstudioconnect. However, I do not know whether this warning is still a problem for this deploy-process.
Thanks a lot in advance!

**Comment by ColinFay** [comment, marked as answer]
Hey, 

Yes, this is a known issue when it comes to publishing to the CRAN, which requires ASCII character only. 
AFAIK, this is a legacy requirement, and almost every computer and server that would run this app will support non ASCII characters.

There are two solutions to this issue here: 

- The one you pointed out, i.e. turning all non ASCII char to their ASCII equivalent. That can be done manually with 

```r
> stringi::stri_escape_unicode("Lélàô ")
[1] "L\\u00e9l\\u00e0\\u00f4 "
```
 
Note that the `{prefixer}` package can do that for you too https://github.com/dreamRs/prefixer/#not-ascii

- Ignore this warning  ¯\_(ツ)_/¯: if ever you only plan on releasing the app to a server and the server doesn't complain about the non ASCII, I suppose there is no harm in not escaping them all. 

See also this chapter for more information  https://r-pkgs.org/r.html#code-cran

Cheers, 
Colin

**Reply by maxl-amberg** [reply]
Ah, I see. 
Thanks a lot (also for mentioning additional resources)!

Best,
Maxl
